tests/unit_tests/test_system_program.py

```diff
@@ -1,6 +1,39 @@
 """Unit tests for solana.system_program."""
 import solana.system_program as sp
 from solana.account import Account
+from solana.publickey import PublicKey
+
+
+def test_decode_instruction_layout():
+    """Test decode a transaction type."""
+    create_account_params = sp.CreateAccountParams(
+        from_pubkey=Account().public_key(),
+        new_account_pubkey=Account().public_key(),
+        lamports=123,
+        space=1,
+        program_id=PublicKey(1),
+    )
+    txn = sp.create_account(create_account_params)
+    assert sp.SYSTEM_INSTRUCTION_LAYOUTS[sp.CREATE_IDX] == sp.decode_instruction_layout(txn.instructions[0])
```

Comment:
I think we should be checking this against `create_account_params`.

Reply:
I don't get this comment. The `decode_instruction_layout` returns an `InstructionLayout` so we should check against a layout right?

Reply:
Oh. This is `decode_instruction_layout`. Nvm.